package/MDAnalysis/analysis/helanal.py

```diff
@@ -248,13 +251,60 @@ def helanal_trajectory(universe, selection="name CA",
     start, end = ca.resids[[0, -1]]
     trajectory = universe.trajectory
 
+    # Validate user supplied begin / end times
+    traj_end_time = trajectory.ts.time + trajectory.totaltime
+
+    if begin is not None:
+        if traj_end_time < begin:
+            # Begin occurs after the end of the trajectory, throw error
+            msg = ("The input begin time ({0} ps) occurs after the end "
+                   "of the trajectory ({1} ps)".format(begin, traj_end_time))
+            raise ValueError(msg)
+        elif trajectory.ts.time > begin:
+            # Begin occurs before trajectory start, warn and reset
+            msg = ("The input begin time ({0} ps) precedes the starting "
+                   "trajectory time --- Setting starting frame to 0".format(
+                    begin))
+            warnings.warn(msg)
+            logger.warning(msg)
+            start_frame = None
+        else:
+            start_frame = int(np.ceil((begin - trajectory.ts.time)
+                                      / trajectory.ts.dt))
+    else:
+        start_frame = None
+
     if finish is not None:
-        if trajectory.ts.time > finish:
-            # you'd be starting with a finish time (in ps) that has already passed or not
-            # available
-            raise FinishTimeException(
-                'The input finish time ({finish} ps) precedes the current trajectory time of {traj_time} ps.'.format(
-                    finish=finish, traj_time=trajectory.time))
+        if (begin is not None) and (begin > finish):
+            # finish occurs before begin time
+            msg = ("The input finish time ({0} ps) precedes the input begin "
+                   "time ({1} ps)".format(finish, begin))
+            raise FinishTimeException(msg)
```

Comment:
While you're at it, could you please get rid of FinishTimeException and just raise a ValueError (because the user provided a wrong value). Makes more sense...